File: reference/springboot/domain-usecases/src/main/java/org/example/domain/usecases/exceptions/NotAuthenticatedException.java
package org.example.domain.usecases.exceptions;

public class NotAuthenticatedException extends RuntimeException {

    protected NotAuthenticatedException() {
        this(null);
    }

    protected NotAuthenticatedException(Throwable cause) {
        super("User must be authenticated to execute this use-case.", cause);
    }
    
}

File: reference/dotnet/Domain/Company.Product.Domain.UseCases/Exceptions/NotAuthenticatedException.cs
using System;

namespace Company.Product.Domain.UseCases.Exceptions
{
    public class NotAuthenticatedException : Exception
    {
        public NotAuthenticatedException()
           : this(null) { }

        public NotAuthenticatedException(Exception innerException)
            : base("User must be authenticated to execute this use-case.", innerException) { }
    }
}